### PR TITLE
fix(person): avoid repeated double last name in some locales

### LINKED ISSUE #1817: Double last names in some locales
### Pre-Checks

- [X] Follow our [Code of Conduct](https://github.com/faker-js/faker/blob/next/CODE_OF_CONDUCT.md).
- [X] Read the [Contributing Guidelines](https://github.com/faker-js/faker/blob/next/CONTRIBUTING.md).
- [X] Read the [docs](https://fakerjs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/faker-js/faker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Faker issue and not related to a combination with another package.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/faker-js/faker/discussions) or join our [Discord Chat Server](https://chat.fakerjs.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.
- [X] I am willing to provide a PR.

### Describe the bug

#1637 has caused a problem in some locales for example `es` which have name patterns like

`'{{person.firstName}} {{person.lastName}} {{person.lastName}}'`

### Minimal reproduction code
```
const {faker} = require("@faker-js/faker")
faker.locale = "es"
faker.person.fullName() //'Anni Cervantes Cervantes'
```

### Additional Context

_No response_

### Environment Info

```shell
System:
    OS: macOS 13.1
    CPU: (8) x64 Apple M1
    Memory: 37.86 MB / 8.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 14.21.2 - /usr/local/bin/node
    npm: 8.19.3 - /usr/local/bin/npm
  Browsers:
    Chrome: 110.0.5481.77
    Firefox: 109.0.1
    Safari: 16.2
```

### Which module system do you use?

- [X] CJS
- [ ] ESM

### Used Package Manager

pnpm

### PR DESCRIPTION
**Pull request by matthewmayer**
fix #1817 

Since faker.mustache will always replace the same placeholder with the same name, tried introducing lastName2

If you have any better ideas for this let me know.

Note this affects #691 (the hyphenated patterns discussed there are already used in some locales)

### CODE REVIEW

File: src/locales/nb_NO/person/name.ts [hunk 1]
```diff
@@ -8,11 +8,11 @@ export default [
     weight: 1,
   },
   {
-    value: '{{person.firstName}} {{person.firstName}} {{person.lastName}}',
+    value: '{{person.firstName}} {{person.firstName}} {{person.lastName2}}',
```
Comment (ST-DDT) [inline src/locales/nb_NO/person/name.ts:11]: This pattern only has a single last name.

**Review by ST-DDT — COMMENTED** [review]
I think this PR tries to fix a problem, without taking the cause of the problem into consideration.

There are two problems here:
1. The patterns seem to express an additional context to the last names that they are unable to express: The person doesn't have a single last name, but two.
2. Some of the patterns seem to modify the exontext of last name to express complex last names such as combined/joined last names.

IMO the real solution to this problem is removing the repeated lastNames from the pattern and actually pass the entire last name to the method directly. (Where would the other last name come from anyway in legal documents? "John" + "Smith" ⇏ "John Smith Foobar")

If there is a locale that actually has two separate last names, then it has to use a different/not yet existing method or parameter set for it. If a locale has some kind of double last name, then it should have them in the locale data itself. Similar to thetussenvoegsel (#1778).

What do you think?

**Comment by matthewmayer** [comment]
I don't think it's quite like tuessenvoegels because there's no "common" double or hyphenated names. In most cultures it's just your mothers and fathers surnames concatenated in some way. So if there are N simple last names there are N^2 compound last names. 

I guess we could add an optional last_name_pattern. Then if provided that would affect both lastName() and fullName()

**Comment by ST-DDT** [comment]
> there's no "common" double or hyphenated names.

It doesn't has to, just have them in our sample set.

> I guess we could add an optional last_name_pattern.

That might be a alternative worth exploring. (We originally planned to remove all optional data access, so we could throw a useful error message instead of returning undefined, but that is something we can keep in mind when implementing this.)

**Comment by matthewmayer** [comment]
> > I guess we could add an optional last_name_pattern.
> 
> That might be a alternative worth exploring. (We originally planned to remove all optional data access, so we could throw a useful error message instead of returning undefined, but that is something we can keep in mind when implementing this.)



i tried out this strategy in https://github.com/faker-js/faker/pull/1819

**Comment by ST-DDT** [comment]
**Team Decision**

Closing in favor of #1819.

**Pull request closed without merging**